pkg/virt-handler/vm.go

```diff
@@ -1272,9 +1272,10 @@ func (d *VirtualMachineController) checkNetworkInterfacesForMigration(vmi *v1.Vi
 		networks[network.Name] = network.DeepCopy()
 	}
 	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
-		if iface.Bridge != nil && networks[iface.Name].Pod != nil {
-			return fmt.Errorf("cannot migrate VMI with a bridge interface connected to a pod network")
+		if iface.Masquerade != nil && networks[iface.Name].Pod != nil {
+			continue
 		}
+		return fmt.Errorf("cannot migrate VMI with the given interface & network, only Masquerade with a Pod network allowed for migration")
```

Comment:
This would return error for secondary (Multus) interfaces with both Masquerade and Bridge. I think you need to have nested ifs:

```
if pod != nil:
    if masquerade:
        continue
    return error
```

Also, would be the default network configured at this point if none was provided by the user? If the user did not specify any network, the list may be empty (unless we add defaults), defaulting to bridge connection to pod.

Reply:
Corrected the nested if and added a test to check this func against a VMI without any networks defined.

Reply:
virt-handler will always see all networks on the VMI. If there is no network, then there is none. The default network is added to the VMI at creation time.

Reply:
@vatsalparekh I think this line would return an error for every secondary interface.

Reply:
@rmohr removed the first error msg now